### ISSUE TITLE
keycloak 22.0.4: desired behavior or regression ? infinite loop login prompt with incognito chrome mode

### ISSUE THREAD

**Issue by benjaminlefevre** [keycloak/keycloak#23840]
### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the [latest nightly release](https://github.com/keycloak/keycloak/releases/tag/nightly)

### Area

oidc

### Describe the bug

Hi guys, I have a behaviour I don't really understand since I upgraded my keycloak in 22.0.4 (everything was working perfectly clear since 19 to 22.0.3).

I have a spa on a domain1. My authentication server keycloak on another domain2.

When I try to login in incognito mode (firefox or chrome), I correcty connect to the app, but I'm logged out end redirected to the login page.

I guess It's related to third party cookies disabled in incognito mode.

Is it normal? why it was working before and suddenly stop to work? What's the solution? Can someone help me? Thank you in advance and bravo for this superb software.

### Version

22.0.4

### Expected behavior

see description

### Actual behavior

see description

### How to Reproduce?

see description

**Comment by benjaminlefevre** [comment]
a little more details: our spa is developed in Angular and uses the library [angular-oauth2-oidc](https://github.com/manfredsteyer/angular-oauth2-oidc) , with the [check of sessions enabled](https://github.com/manfredsteyer/angular-oauth2-oidc)

**Comment by benjaminlefevre** [comment]
maybe the issue is on our side and revelead by a change on keycloak side... probably... currenttly we logout the user when session error event type is triggered... the error event occurs now since 22.0.4 when third party cookies are disabled

**Comment by jonkoops** [comment]
> the error event occurs now since 22.0.4 when third party cookies are disabled

Where exactly are you seeing the `error` event being triggered? We did add a new `error` state when the login status is being checked and we [cannot guarantee](https://github.com/keycloak/keycloak/blob/9df1c781eb34e2512a49414d016ab00e496f8d51/services/src/main/resources/org/keycloak/protocol/oidc/endpoints/login-status-iframe.html#L38-L40) there is access to 3rd-party cookies, so it's likely because of this change.

We could change this to always return a `unchanged` state, which would prevent exceptions in 3rd-party libraries such as `angular-oauth2-oidc`. However, we would essentially be lying as we cannot actually determine the session state, and by extension changes to it.

@manfredsteyer as the author of this library, I would be interested in your take on this. @panva this discussion might also be relevant for you.

**Comment by panva** [comment]
I have never supported OpenID Connect Session Management in my client libraries. And I have removed the server support for it years ago as well. 

This mechanism was unfortunately known to be DOA for several years. Some of us even objected to it getting to Final because of it.

**Comment by benjaminlefevre** [comment]
@jonkoops thank you for your answer. Yes I exactly talk about the change you mention. The "error" returned when no access to 3rd party cookies. 

It's fine to me, because as you said "we would essentially be lying". I just have to ignore this kind of event. 

angular-oauth2-oid is converting the 'error' to 'session_error' event when checking state, which leads to an automatic logout with our code. We can just ignore and act like nothing happened. For information, we detected this issue, since one of our customer has disabled third party cookies on all of its workstations.

**Comment by jonkoops** [comment]
> This mechanism was unfortunately known to be DOA for several years. Some of us even objected to it getting to Final because of it.

Yeah, I have read that little foot-note in the spec. It's unfortunate, but it's indeed way to hard to get this done right. Ignoring the fact that this will just be dead the second all browsers drop most cookie support in 2014.

**Comment by jonkoops** [comment]
> angular-oauth2-oid is converting the 'error' to 'session_error' event when checking state, which leads to an automatic logout with our code. We can just ignore and act like nothing happened.

Yeah, I think that is the best approach you can have here, try to fall back gracefully where possible. I'll consider this issue resolved then.
